File: macos/Tests/VoiceTyperTests/TextPostprocessorTests.swift
import XCTest
@testable import VoiceTyper

final class TextPostprocessorTests: XCTestCase {
    func testStripsRichTags() {
        let raw = "<|zh|><|NEUTRAL|><|Speech|><|withitn|>你好世界"
        XCTAssertEqual(TextPostprocessor.process(raw), "你好世界")
    }

    func testSentencePieceSpaceMarkerBecomesSpace() {
        XCTAssertEqual(TextPostprocessor.process("▁hello▁world"), "hello world")
    }

    func testPureSubstantivelessResultIsDropped() {
        XCTAssertEqual(TextPostprocessor.process("。"), "")
        XCTAssertEqual(TextPostprocessor.process("<|zh|><|NEUTRAL|><|Speech|><|withitn|>。"), "")
    }

    func testRemovesSpaceAfterCJKBeforeLatin() {
        XCTAssertEqual(TextPostprocessor.process("你好▁issue"), "你好issue")
    }

    func testRemovesSpaceBeforeCJKAfterLatin() {
        XCTAssertEqual(TextPostprocessor.process("issue▁你好"), "issue你好")
    }

    func testKeepsSpaceBetweenLatinWords() {
        XCTAssertEqual(TextPostprocessor.process("audio▁driven▁text"), "audio driven text")
    }

    func testCollapsesRepeatedWhitespace() {
        XCTAssertEqual(TextPostprocessor.process("hello   world"), "hello world")
    }

    func testKeepsJapaneseAndKoreanAsSubstantive() {
        XCTAssertEqual(TextPostprocessor.process("こんにちは"), "こんにちは")
        XCTAssertEqual(TextPostprocessor.process("안녕하세요"), "안녕하세요")
    }
}

File: windows/Tests/VoiceTyper.Tests/TextPostprocessorTests.cs
using VoiceTyper.Asr;
using Xunit;

namespace VoiceTyper.Tests;

public class TextPostprocessorTests
{
    [Fact]
    public void StripsRichTags()
    {
        Assert.Equal("你好", TextPostprocessor.Process("<|zh|><|NEUTRAL|><|Speech|><|withitn|>你好"));
    }

    [Fact]
    public void ConvertsSentencePieceMarkerToSpace_ThenCollapsesAroundCjk()
    {
        var result = TextPostprocessor.Process("你好▁world▁再见");
        Assert.Equal("你好world再见", result);
    }

    [Fact]
    public void DropsPureWhitespaceOrPunctuationOnlyResult()
    {
        Assert.Equal("", TextPostprocessor.Process("<|zh|><|NEUTRAL|><|Speech|><|withitn|>。"));
    }

    [Fact]
    public void KeepsSubstantiveEnglishText()
    {
        Assert.Equal("hello world", TextPostprocessor.Process("hello world"));
    }

    [Fact]
    public void CollapsesRepeatedWhitespace()
    {
        Assert.Equal("a b", TextPostprocessor.Process("a    b"));
    }

    [Fact]
    public void EmptyInput_ReturnsEmpty()
    {
        Assert.Equal("", TextPostprocessor.Process(""));
    }
}
